GoogleForCreators/web-stories-wp
# Media: Transcode videos inserted from WordPress media modal

## Context



## Summary

All for existing video that can be transcode to be shown in the default WordPress media picker. Once the video is select, insert a placeholder into content. From there, generate poster image and start the transcoding process. 

Some video types, like mpg, the video will not have height and width data, so a default height and width will need to be set. 

The allow list of transcodable mime type will have to moved to PHP, as it is needed for the jetpack integration. A filter is not required for this list of transcodable mime types. 

This PR will likely be effected by https://github.com/google/web-stories-wp/pull/7699 https://github.com/google/web-stories-wp/pull/7691 https://github.com/google/web-stories-wp/pull/7317. I would like to see all these PRs merged first, so this PR can be rebased and retested. 

## Relevant Technical Choices



## To-do

- [ ] Add some kind of test

## User-facing changes

#### Upload MOV ( that generates height and width )

https://user-images.githubusercontent.com/237508/119982431-bc5b8680-bfb6-11eb-918f-a6bd0aef6f1c.mov

#### Upload mpg  ( that generates doesn't height and width )
https://user-images.githubusercontent.com/237508/119982391-afd72e00-bfb6-11eb-8905-1cf89776117d.mov


## Testing Instructions



### QA






This PR can be tested by following these steps:

1. Enable Video optimization in media picker feature flag. 
2. Create new story. 
3. Click upload button. 
4. Upload, .mov or .mpg file. 
5. Wait for file to upload. 
6. Once complete, click insert in to page. 
7. See video is inserted. 
8. See placeholder is inserted.
9. See update once poster is generated, element may change size and position. 
10. See that video controls are not visible. 
11. Once trancode is finished, video should appear in sidebar and video should be playable. 

This functionality should be work, if users disables transcoding options. See dashboard -> settings. 

1. Enable Video optimization in media picker feature flag. 
2. Go to media in sidebar. 
3. Upload .mov or .mpg file by clicking "add new"
2. Create new story. 
3. Click upload button. 
5. Click on media library. 
6. Once complete, click insert in to page. 
7. See video is inserted. 
8. See placeholder is inserted.
9. See update once poster is generated, element may change size and position. 
10. See that video controls are not visible. 
11. Once trancode is finished, video should appear in sidebar and video should be playable. 

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #7480
